script/vendor-mimes

```diff
@@ -23,13 +48,27 @@ data.each do |mime, meta|
   next if extensions.empty?
   mimes[mime] = [] if mimes[mime].nil?
   mimes[mime].concat extensions
+
+  # Extract mime-types with "charset" metadata
+  charset_data[mime] = meta["charset"] if meta.key?("charset")
 end
 
+log_info "Formatting primary hash and writing to file.."
 strlen = mimes.keys.max_by(&:length).length
 output = ""
 output << "# Woah there. Do not edit this file directly.\n"
 output << "# This file is generated automatically by script/vendor-mimes.\n\n"
 mimes = mimes.sort_by { |k,v| k }
 output << mimes.map { |mime,extensions| "#{mime.ljust(strlen)} #{extensions.sort.join(" ")}" }.join("\n")
 
+config = File.expand_path "../lib/jekyll/mime.types", __dir__
 File.write(config, output)
+log_info "Done! See: #{config.inspect.white}"
+
+# --- Generate JSON file from charset_data ----
+puts
+
+log_info "Dumping miimetype-charset mapping as JSON.."
```

Comment:
```suggestion
log_info "Dumping mimetype-charset mapping as JSON.."
```

Reply:
Thank you so very much I appreciate  everyone's help